### PR TITLE
Add Affine Cipher

### PR DESCRIPTION
**Pull request by shidian00**
[Open in Gitpod](https://gitpod.io/from-referrer/)&nbsp;[know more](https://www.gitpod.io/docs/pull-requests/)

### Describe your change:

- [x] Add an algorithm?
- [ ] Fix a bug or typo in an existing algorithm?
- [ ] Documentation change?

### Checklist:

- [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Javascript/blob/master/CONTRIBUTING.md).
- [x] This pull request is all my own work -- I have not plagiarized.
- [x] I know that pull requests will not be merged if they fail the automated tests.
- [x] This PR only changes one algorithm file. To ease review, please open separate PRs for separate algorithms.
- [x] All new JavaScript files are placed inside an existing directory.
- [x] All filenames should use the UpperCamelCase (PascalCase) style. There should be no spaces in filenames.
      **Example:**`UserProfile.js` is allowed but `userprofile.js`,`Userprofile.js`,`user-Profile.js`,`userProfile.js` are not
- [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
- [ ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: Ciphers/AffineCipher.js [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+ * @param {Number} a - A coefficient
+ * @param {Number} b - B coefficient
+ * @return {String} result - Encrypted string
+ */
+function encrypt (str, a, b) {
+  let result = ''
+  if (isCorrectFormat(str, a, b)) {
+    for (let x = 0; x < str.length; x++) {
+      if (key.indexOf(str[x].toUpperCase()) === -1) result += '-1' + ' '
+      else result += mod(a * key.indexOf(str[x].toUpperCase()) + b, 26) + ' '
```
Comment (appgurueu) [inline Ciphers/AffineCipher.js:63]: Consider using `str[x].toUpperCase().charCodeAt(0) - ("A").charCodeAt(0)`. This runs in constant time whereas `key.indexOf` runs in linear time in the size of the alphabet.

File: Ciphers/AffineCipher.js [hunk 2]
```diff
@@ -0,0 +1,88 @@
+
```
Comment (appgurueu) [inline Ciphers/AffineCipher.js:1]: Leading spacing.

File: Ciphers/AffineCipher.js (hunk 1 above)
Comment (shidian00) [reply Ciphers/AffineCipher.js]: Thank you for reviewing my code. I have replace indexOf with your recommendation

File: Ciphers/AffineCipher.js [hunk 3]
```diff
@@ -47,6 +47,14 @@ function isCorrectFormat (str, a, b) {
 
   return true
 }
+/**
+ * Find character index based on ASCII order
+ * @param {String} char - Character index to be found
+ * @return {Boolean} Chracter index
+ */
+function findCharIndex (char) {
+  return char.toUpperCase().charCodeAt(0) - ('A').charCodeAt(0)
```
Comment (appgurueu) [inline Ciphers/AffineCipher.js:55]: NVM, this isn't Lua, you don't need the parenthesis around `'A'`.

**Review by appgurueu — APPROVED** [review]

**Review by raklaptudirm — APPROVED** [review]

**Pull request merged**